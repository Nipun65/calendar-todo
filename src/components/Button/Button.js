import { handleFunc } from "../Handlers/Handlers";
import "./Button.module.css";

const Button = ({ onClick, classes, disabled, textContent }) => {
  return (
    <button
      onClick={() => handleFunc(onClick)}
      className={classes}
      disabled={disabled}
    >
      {textContent}
    </button>
  );
};

export default Button;
