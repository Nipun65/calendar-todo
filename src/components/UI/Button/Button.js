import './Button.module.css';

function Button({ onClick, classes, disabled, textContent }) {
  const clickHandler = () => {
    onClick();
  };
  return (
    <button
      onClick={clickHandler}
      className={classes}
      disabled={disabled}
      type="button"
    >
      {textContent}
    </button>
  );
}

export default Button;
