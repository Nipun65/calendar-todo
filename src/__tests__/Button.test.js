import renderer from 'react-test-renderer';
import Button from '../components/UI/Button/Button';

it('should match the snapshot', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
