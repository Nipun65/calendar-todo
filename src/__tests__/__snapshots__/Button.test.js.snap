// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<button
  className="undefined is-clickable"
  onClick={[Function]}
  type="button"
/>
`;
